src/features/boundary_selector/atoms/boundaryMarkerAtom.ts

```diff
@@ -100,19 +97,38 @@ export const boundaryMarkerAtom = createAtom(
 
           const markerData = getSelectorWithOptions(
             options,
+
             // onOptionSelect:
             (boundaryId) => {
               if (!featureCollection) return;
+              const selectedFeature = featureCollection.features.find(
```

Comment:
also it's possibly null here, is there any sense to process this case?

Reply:
`selectedFeature` can't be null here, featureCollection.features is array of features (they're objects of type GEOJson.Feature). Have you meant something else?

Reply:
I just ment that .find() can or cant find smth
if it can't -> selectedFeature will be null

Reply:
I see, it could be `undefined` if nothing found, but there's `!` in the end, meaning we're shure it will find something. I wrote that `!` because previous code implied that and because `boundaryId` and `featureCollection` that we are comparing against comes from the same source